File: server-pedidos-java/src/main/java/br/edu/utfpr/serverpedidosjava/entity/Produto.java
package br.edu.utfpr.serverpedidosjava.entity;

import jakarta.persistence.*;

import java.math.BigDecimal;

@Entity
@Table(name = "meu_produto")
public class Produto {

    @Id
    @GeneratedValue
    private Integer id;
    @Column(name = "nome_produto", nullable = false, length = 100)
    private String nome;
    @Column(name = "valor_venda", nullable = false, precision = 15, scale = 2)
    private BigDecimal preco;
    @ManyToOne(optional = false)
    @JoinColumn(name = "id_categoria")
    private Categoria categoria;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public BigDecimal getPreco() {
        return preco;
    }

    public void setPreco(BigDecimal preco) {
        this.preco = preco;
    }

    public Categoria getCategoria() {
        return categoria;
    }

    public void setCategoria(Categoria categoria) {
        this.categoria = categoria;
    }
}

File: server-pedidos/src/main/kotlin/br/edu/utfpr/serverpedidos/entity/Produto.kt
package br.edu.utfpr.serverpedidos.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Positive
import jakarta.validation.constraints.Size
import java.math.BigDecimal

@Entity
@Table(name = "meu_produto")
class Produto(
    @Id
    @GeneratedValue
    var id: Int,
    @field:NotBlank(message = "{nome.notblank}")
    @field:Size(max = 100, message = "{nome.size}")
    @Column(name = "nome_produto", nullable = false, length = 100)
    var nome: String = "",
    @field:Positive(message = "{preco.positive}")
    @Column(name = "valor_venda", nullable = false, precision = 15, scale = 2)
    var preco: BigDecimal = BigDecimal.ZERO,
    @field:NotNull(message = "{categoria.notnull}")
    @ManyToOne(optional = false)
    @JoinColumn(name = "id_categoria")
    var categoria: Categoria? = null
)
